src/bootstrap/network.js

```diff
@@ -1,16 +1,25 @@
 import web3 from "./web3";
 import { ERRORS } from "../utils/errorCodes";
+import {
+  PROVIDER_TYPE_METAMASK,
+  PROVIDER_TYPE_MIST,
+} from '../constants';
 
-export function checkConnectivity() {
-  return (() => {
+export const checkConnectivity = () =>
+  (() => {
     return new Promise((resolve, reject) => {
-      const tid = setTimeout(() => reject(ERRORS.NO_CONNECTION), 2000);
-      web3.version.getNode((error, result) => {
+      const tid = setTimeout(() => reject(ERRORS.NO_CONNECTION), 4000);
+      if (!window.mist) {
+        web3.version.getNode((error, result) => {
+          clearTimeout(tid);
```

Comment:
Probably this `clearTimeout(tid)` can be extracted outside the  if / else block because you do it in both cases anyway